src/components/TableChats.vue

```diff
@@ -0,0 +1,193 @@
+<template>
+  <section class="table-chats">
+    <header class="table-chats__tab-buttons">
+      <button
+        v-for="(tab, index) in tabs"
+        :key="index"
+        @click="changeTab(index)"
+        :class="{ active: activeTab === index }"
+        class="table-chats__tab-button"
+      >
+        {{ tab }}
+      </button>
+    </header>
+    <table class="table-chats__table">
+      <thead class="table-chats__table-head">
+        <tr>
+          <th v-for="header in headers" :key="header">{{ header }}</th>
+        </tr>
+      </thead>
+      <tbody class="table-chats__table-body">
+        <tr
+          v-for="(row, rowIndex) in tableData"
+          :key="rowIndex"
+          class="table-chats__table-row"
+        >
+          <td v-for="(value, colIndex) in row" :key="colIndex">{{ value }}</td>
+        </tr>
+      </tbody>
+    </table>
+    <hr class="table-chats__separator" />
+    <footer class="table-chats__bottom">
+      <TablePagination
+        v-model="chatsCurrentPage"
+        :value="chatsCurrentPage"
+        :count="chatsCount"
+        :countPages="
+          getTotalPages(Object.keys(this.data.chats.status)[this.activeTab])
+        "
+        :limit="chatsLimit"
+        @updatePage="updatePage"
+      />
+    </footer>
+  </section>
+</template>
+
+<script>
+import TablePagination from '@/components/TablePagination.vue';
+
+export default {
+  name: 'TableChats',
+
+  components: {
+    TablePagination,
+  },
+
+  data() {
+    return {
+      tableData: [],
+      tabData: [],
+      activeTab: 0,
+      chatsCurrentPage: 1,
+      chatsCount: 0,
+      chatsCountPages: 0,
+      chatsLimit: 10,
+    };
+  },
+
+  props: {
+    tabs: {
+      type: Array,
+      default: () => ['Aguardando', 'Em progresso', 'Encerrado'],
+    },
+    headers: {
+      type: Array,
+      default: () => [
+        'Contato',
+        'Agente',
+        'Horário de início',
+        'Tempo de interação',
+      ],
+    },
+    data: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  methods: {
+    changeTab(index) {
+      this.activeTab = index;
+      this.chatsCurrentPage = 1;
+    },
+
+    updateTableData(tabs) {
+      const tabData = this.data.chats.status[tabs];
+      const startIndex = (this.chatsCurrentPage - 1) * this.chatsLimit;
+      const endIndex = startIndex + this.chatsLimit;
+
+      this.chatsCount = tabData.length;
+      this.tableData = tabData.slice(startIndex, endIndex).map((item) => ({
+        contact: item.contact,
+        agent: item.agent,
+        start_time: item.start_time,
+        interaction_time: item.interaction_time,
+      }));
+    },
+
+    getTotalPages(tabs) {
+      const tabData = this.data.chats.status[tabs];
+      if (tabData && tabData.length) {
+        return Math.ceil(tabData.length / this.chatsLimit);
```

Comment:
It is preferable to transform `Math.ceil(tabData.length / this.chatsLimit)` into a variable and return the variable, for better readability and understanding of what this calculation would be